std/src/main/scala/algebra/std/map.scala

```diff
@@ -26,33 +50,80 @@ class MapEq[K, V](implicit ev: Eq[V], a: AdditiveMonoid[V]) extends Eq[Map[K, V]
     check(x, y) && check(y, x)
```

Comment:
If scala really had haskell's zero cast newtype (AnyVal feels like 1/3 the job), the wrapper approach + no orphan instances would be my preferred philosophy. Even with the weaknesses of wrappers, we use them in a few places in algebird to control typeclass selection. I don't see us doing it anywhere in algebra so far, perhaps due to the less frequent use of default typeclass instances.